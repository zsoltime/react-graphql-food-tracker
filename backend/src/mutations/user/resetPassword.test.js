'use strict';

const resetPassword = require('./resetPassword');

describe('resetPassword mutation', () => {
  it('throws error if token is not valid', () => {
    const args = { password: 'test1234', token: 'invalidToken' };
    const mockedContext = {
      db: {
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([])),
        },
      },
    };

    expect(
      resetPassword(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error if password is not valid', () => {
    const args = { password: 'test123', token: 'invalidToken' };
    const mockedContext = {
      db: {
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([{ email: 'valid@email.io' }])),
        },
      },
    };

    expect(
      resetPassword(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
