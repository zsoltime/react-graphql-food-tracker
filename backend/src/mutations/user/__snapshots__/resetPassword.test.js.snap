// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetPassword mutation throws error if password is not valid 1`] = `"Password must be at least 8 characters"`;

exports[`resetPassword mutation throws error if token is not valid 1`] = `"Token is either invalid or expired"`;
